### DISCUSSION TITLE
how can I export diagram with full size background image?

### DISCUSSION THREAD

**Discussion by vyshak-pr** [clientIO/joint#1951] [Q&A]
This is my  diagram[](url)
![diagram](https://user-images.githubusercontent.com/120721238/208423375-95362742-4f25-4d3a-897f-42566b599c38.png)

while I exporting the diagram it is not showing the full background image
![png format](https://user-images.githubusercontent.com/120721238/208424073-8be56a41-155d-4ee6-acea-1c8e9ab74fc8.png)

how can I solve the issue?

**Comment by kumilingus** [comment]
The answer is the same as in https://github.com/clientIO/joint/issues/1502.

You just need to fix the size of the image with `toPNG()` `area` option. For instance:

```js
const imageBBox = new g.Rect(0, 0, 1000, 1000);

const imageVel = V('image')
    .attr('xlink:href', 'assets/image-icon2.svg')
    .attr(imageBBox.toJSON());

paper.getLayerNode('back').appendChild(imageVel.node);

paper.toPNG((dataURL) => {
    imageVel.remove();
    /*... do something with the image... */
}, {
    padding: 10,
    useComputedStyles: false,
    area: imageBBox
});
```
The size of `imageBBox` depends on how did you define the background image. How did you define it?
